src/libstrongswan/plugins/openxpki/openxpki_ocsp_responder.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2023 Andreas Steffen, strongSec GmbH
+ *
+ * Copyright (C) secunet Security Networks AG
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.  See <http://www.fsf.org/copyleft/gpl.txt>.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ */
+
+#include <library.h>
+#include <utils/debug.h>
+
+#include "openxpki_ocsp_responder.h"
+
+
+typedef struct private_openxpki_ocsp_responder_t private_openxpki_ocsp_responder_t;
+
+/**
+ * Private Data of a openxpki_ocsp_responder_t object.
+ */
+struct private_openxpki_ocsp_responder_t {
+
+	/**
+	 * Public data
+	 */
+	openxpki_ocsp_responder_t public;
+
+	/**
+	 * OpenXPKI certificate database
+	 */
+	database_t *db;
+
+};
+
+
+METHOD(ocsp_responder_t, get_status, cert_validation_t,
+	private_openxpki_ocsp_responder_t *this, certificate_t *cacert,
+	chunk_t serial_number, time_t *revocation_time, crl_reason_t *reason)
+{
+	cert_validation_t validation = VALIDATION_FAILED;
+	int rev_time;
+	char *status, *reason_code;
+	char serial[50], authKeyId[60];
+	chunk_t subjectKeyId;
+	public_key_t *public;
+	enumerator_t *e;
+	bool success;
+
+	/* convert serialNumber from hex to decimal required for the DB query */
+	chunk_to_dec(serial_number, serial);
```

Comment:
Added a length check equivalent to the decimal digits computation in chunk_to_dec()

Reply:
Why use 2.4083 here and 2.41 there? But actually, instead of calculating this, a simple check for `serial_number.len > 20` and a comment on the calculation and the 49 character limit would be a lot simpler.